src/main/java/core/basesyntax/stategy/handler/SupplyOperation.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.stategy.handler;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void process(FruitTransaction transaction) {
+        String fruit = transaction.getFruitName();
+        int quantity = transaction.getQuantity();
+        if (quantity < 0) {
+            throw new IllegalArgumentException("Invalid quantity: " + quantity);
+        }
+        Storage.fruits.merge(fruit, quantity, Integer::sum);
```

Comment:
Ensure that the class follows the encapsulation principle. The 'fruits' field in the Storage class should be private and accessed via methods to adhere to encapsulation.